olive/hardware/accelerator.py

```diff
@@ -146,12 +146,36 @@ def infer_accelerators_from_execution_providers(execution_providers: List[str]):
                     mapped_devices.append(inferred_device[0])
         return mapped_devices if mapped_devices else None
 
+    @staticmethod
+    def infer_single_device_from_execution_providers(execution_providers: List[str]) -> str:
+        if not execution_providers:
+            return None
 
-def infer_accelerators_for_system_config(system_config: "SystemConfig") -> "SystemConfig":
-    """If the accelerators are not specified, infer the accelerators. otherwise, return the original system config."""
+        if execution_providers == ["CPUExecutionProvider"]:
+            inferred_accelerators = ["cpu"]
+        else:
+            inferred_accelerators = AcceleratorLookup.infer_devices_from_execution_providers(execution_providers)
+            assert (
+                inferred_accelerators
+            ), f"Cannot infer the accelerators from the execution providers {execution_providers}."
+            assert len(inferred_accelerators) == 1, (
+                f"Cannot infer the accelerators from the execution providers {execution_providers}. "
```

Comment:
Ditto. Also please remind user that they can set the device manually to mitigate this issue.